### ISSUE TITLE
Edit default init files

### ISSUE THREAD

**Issue by LarsDR** [tongueroo/ufo#26]
Hi!

Is it somehow possible to change the templates for the files, that the `ufo init`-command generates?

Most of my services follow the same overall structure, which means I'll have to perform the same changes every time I initialize a new UFO setup.

Are the "default setup files" stored somewhere accessible, where they're possible to change?

I'm on a Mac, by the way :-)

**Comment by tongueroo** [comment]
@LarsDR Interesting. That is annoying to have to edit the templates every time. Will add this as a feature. 

As a workaround, you can edit the templates in the gem itself. To find the template directory on your system, you can use the `gem which ufo` command.  Example:

```
$ gem which ufo
/Users/tung/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/ufo-3.2.1/lib/ufo.rb
```

From there, the template folder is one level up:

```
$ ls -a /Users/tung/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/ufo-3.2.1/lib/template
.env       .ufo       Dockerfile bin
```

Here's a command that will open it up in sublime in one go:

```
subl -a $(dirname $(gem which ufo))/template
```

Some cons:

* whenever ufo is upgraded, you'll lose your changes.
* the custom templates are not sharable. 

Thinking the feature will address these. Hope the workaround helps for now! 🎉

**Comment by LarsDR** [comment]
Hey Tong!

Thanks (once again) for your fast reply :-)
After posting this, I got thinking about another enhancement of my suggestion: Being able to create different 'templates'.

Maybe the 'default' configurations could be placed in `~/.ufo/default`, but then it could be possible to add `~/.ufo/tpl-a/` as well. In my initialization command, I could then do something like `ufo init --app=my-app --image=my-image --tpl=tpl-a`.

What do you think? :-)

**Comment by tongueroo** [comment]
@LarsDR Like it! Was thinking the same thing. 

Would be also cool to be able to specify a url with the `--template` optoin.  It would result in the template getting downloaded in `~/.ufo/templates/github-username/gitub-repo`, like:

```
ufo init --app=my-app --image=my-image --template=tpl-a
ufo init --app=my-app --image=my-image --template=https://github.com/username/ufo-template
ufo init --app=my-app --image=my-image --template=username/ufo-template # also works
```

**Comment by tongueroo** [comment]
@LarsDR @azgul #27 handles this. Merged and released in version 3.3.0. https://github.com/tongueroo/ufo/blob/master/CHANGELOG.md

Docs on how to use the `--template` and `--template-mode` options have been added at the bottom of the ufo init reference: http://ufoships.com/reference/ufo-init/

There are 2 template modes: additive and replace. 

* With the additive mode, the custom `~/.ufo/templates` folder gets "merged" with the [default template](https://github.com/tongueroo/ufo/tree/master/lib/template). 
* With the replace mode, only your custom `~/.ufo/templates` folder files get used.
